### PR TITLE
Fix bot py3.9

### PR DESCRIPTION
**Pull request by pinin4fjords**
This is an attempted fix to the bot's last PR in https://github.com/conda-forge/multicore-tsne-feedstock/pull/5. Changes:

 - Pin cmake to a version compatible with the setup.py for this package.
 - ~~Make numpy unpinned as per source pip package.~~
 - Reduce tests to import only (there was a pinned dep there that was causing conflicts, and it's my understanding that tests at that level should not be part of Conda recipes anyway). 

Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by pinin4fjords** [comment]
Could I get review @conda-forge/multicore-tsne?

**Comment by chrisburr** [comment]
> Make numpy unpinned as per source pip package

This is actually needed when building against the numpy C API in case they ever release a version 2 with an ABI break: https://conda-forge.org/docs/maintainer/knowledge_base.html#linking-numpy

**Comment by pinin4fjords** [comment]
Thanks @chrisburr I've reinstated it- I think it was the rigid dep in the tests that was causing the main problem anyway.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,31 +10,26 @@ source:
...
   script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
 
 requirements:
   build:
     - {{ compiler('cxx') }}
-    - cmake
+    - cmake <3.20
   host:
-    - python
+    - python =3.7
```
Comment (chrisburr) [inline recipe/meta.yaml]:
```suggestion
    - python
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,31 +10,26 @@ source:
...
     - pip
-    - numpy
+    - {{ pin_compatible('numpy') }}
     - cffi >=1.11.5
   run:
-    - python >=2.7
+    - python =3.7
```
Comment (chrisburr) [inline recipe/meta.yaml]:
```suggestion
    - python
```

**Review by chrisburr — COMMENTED** [review]
Ah this is wrong for compiled code, python shouldn't be constrained and the recipe then needs to be rerendered

**Comment by chrisburr** [comment]
@conda-forge-admin, please rerender

**Comment by pinin4fjords** [comment]
@chrisburr sorry, I hadn't intended to pin python, that was from my local testing, shouldn't have pushed it.

**Comment by pinin4fjords** [comment]
@conda-forge/core would you be able to review this?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,31 +10,26 @@ source:
...
 requirements:
   build:
     - {{ compiler('cxx') }}
-    - cmake
+    - cmake <3.20
   host:
     - python
     - pip
-    - numpy
+    - {{ pin_compatible('numpy') }}
```
Comment (xhochy) [inline recipe/meta.yaml]: We shouldn't have `pin_compatible` in `host`
```suggestion
    - numpy
```

Comment (pinin4fjords) [reply recipe/meta.yaml]: @xhochy that seems to have broken things for Py3.9. Suggestions?

**Comment by mbargull** [comment]
@conda-forge-adming, please rerender

**Comment by mbargull** [comment]
@conda-forge-admin, please rerender

**Comment by pinin4fjords** [comment]
Thanks @mbargull @xhochy - are we good to go now?

**Comment by xhochy** [comment]
@pinin4fjords Can you add yourself as a maintainer? Then I would merge immediately, otherwise we have a grace period for the original maintainer to react.

**Comment by xhochy** [comment]
@conda-forge-admin, please rerender

**Pull request merged**